numba/tests/test_interpreter.py

```diff
@@ -618,3 +620,32 @@ def inline_func(flag):
         a = py_func(False)
         b = cfunc(False)
         self.assertEqual(a, b)
+
+
+class TestListExtendInStarArgNonTupleIterable(MemoryLeakMixin, TestCase):
+    """Test `fn(pos_arg0, pos_arg1, *args)` where args is a non-tuple iterable.
+
+    Python will generate LIST_EXTEND bytecode to combine the positional
+    arguments with the `*args`.
+
+    See #8059
+
+    NOTE: At the moment, there are no meaningful test for NoPython because the
+    lack of support for `tuple(iterable)` for most iterable types.
+    """
+    def test_list_extend_forceobj(self):
+        def consumer(*x):
+            return x
+
+        @jit(forceobj=True)
+        def foo(x):
+            return consumer(1, 2, *x)
+
+        got = foo("ijo")
+        expect = foo.py_func("ijo")
+        self.assertEqual(got, (1, 2, "i", "j", "o"))
```

Comment:
Whilst not strictly needed given the check below, am happy to leave this as-is to offer visual confirmation of what is expected.